google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ReadWriteTransaction.java

```diff
@@ -431,25 +434,27 @@ private ApiFuture<ResultSetStats> internalExecuteUpdateAsync(
                                 StatementExecutionStep.EXECUTE_STATEMENT,
                                 ReadWriteTransaction.this);
 
-                        ResultSetStats updateCount;
+                        Tuple<Long, ResultSet> result;
+                        long updateCount;
                         if (analyzeMode == AnalyzeMode.NONE) {
                           updateCount =
-                              ResultSetStats.newBuilder()
-                                  .setRowCountExact(
-                                      get(txContextFuture)
-                                          .executeUpdate(update.getStatement(), options))
-                                  .build();
+                              get(txContextFuture).executeUpdate(update.getStatement(), options);
+                          result = Tuple.of(updateCount, null);
                         } else {
-                          updateCount =
+                          ResultSet resultSet =
                               get(txContextFuture)
-                                  .analyzeUpdate(
+                                  .analyzeUpdateStatement(
                                       update.getStatement(),
                                       analyzeMode.getQueryAnalyzeMode(),
                                       options);
+                          updateCount =
+                              resultSet.getStats() == null
```

Comment:
When will `resultSet.getStats()` be `null`? Should we assume `updateCount` as `0` in that case?

Reply:
In this case it should never be null, but IntelliJ complains because `getStats()` is annotated as `@Nullable`. I've changed it to use a `requireNonNull` method call instead, which also silences the warning in IntelliJ.

Reply:
Got it, thanks!